tailcfg/tailcfg.go

```diff
@@ -1897,39 +1892,88 @@ type TKAInitFinishRequest struct {
-// WantSync is false.
-//
-// If control has updates that build off the head hash reported by the
-// node, they are simply transmitted in Updates (avoiding the more
-// expensive synchronization process).
-//
-// In all other cases, WantSync is set to true, and the node is expected
-// to reach out to control separately to synchronize.
-type TKAMapResponse struct {
-	// Updates is any AUMs that control believes the node should apply.
-	Updates []tkatype.MarshaledAUM `json:",omitempty"`
-
-	// WantSync is set by control to request the node complete AUM
-	// synchronization.
+// TKAInfo encodes controls' view of tailnet key authority (TKA) state.
+// This information is transmitted as part of the MapResponse.
+type TKAInfo struct {
+	// Head describes the hash of the latest AUM applied to the authority.
+	// Head is encoded as tka.AUMHash.MarshalText()
 	//
-	// TODO(tom): Implement AUM synchronization, probably as noise endpoints
+	// If the Head state differs to that known locally, the node should perform
+	// synchronization via a separate RPC.
+	//
+	// TODO(tom): Implement AUM synchronization as noise endpoints
 	// /machine/tka/sync/offer & /machine/tka/sync/send.
-	WantSync bool `json:",omitempty"`
+	Head string `json:",omitempty"`
+
+	// Enabled describes whether the tailnet key authority should be enabled.
+	//  - False: TKA should be disabled. If it is enabled locally, the node
+	//    should reach out to fetch a disablement secret. If the disablement
+	//    secret verifies, then the node should then disable TKA locally.
+	//  - True: TKA should be enabled. If it is not, the node should reach
+	//    out via separate RPC to fetch the genesis AUM and initialize.
+	//
+	// TODO(tom): Implement /machine/tka/boostrap as a noise endpoint, to
+	// communicate the genesis AUM & any disablement secrets.
+	Enabled bool `json:",omitempty"`
+}
+
+// TKABootstrapRequest is sent by a node to get information necessary for
+// enabling or disabling the tailnet key authority.
+type TKABootstrapRequest struct {
+	// Head represents the nodes' head AUMHash (tka.Authority.Head()), if
+	// network lock is enabled.
+	Head string
+}
+
+// TKABootstrapResponse encodes values necessary to enable or disable
+// the tailnet key authority (TKA).
+type TKABootstrapResponse struct {
+	// GenesisAUM encodes the initial AUM necessary to initialize TKA.
+	GenesisAUM tkatype.MarshaledAUM `json:",omitempty"`
+
+	// DisablementSecret encodes a secret necessary to disable TKA.
+	DisablementSecret []byte `json:",omitempty"`
+}
+
+// TKASyncOfferRequest encodes a request to synchronize tailnet key authority
+// state (TKA). Values of type tka.AUMHash are encoded as strings in their
+// MarshalText() form.
+type TKASyncOfferRequest struct {
+	// Head represents the nodes' head AUMHash (tka.Authority.Head()). This
+	// corresponds to tka.SyncOffer.Head.
+	Head string
+	// Ancestors represents selection of ancestor AUMHash values ascending
```

Comment:
the selection?

but I don't quite follow this so maybe I'm confused

Reply:
A selection (2nd ancestor, 4th ancestor, 8th ancestor, 16th ancestor etc) of ancestor hashes are sent to help compute what we call in the design a 'tail intersection'. This is all a mirror of the fields of `tka.SyncOffer`.